internals/daemon/daemon.go

```diff
@@ -224,35 +165,40 @@ func (c *Command) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	switch c.canAccess(r, user) {
-	case accessOK:
-		// nothing
-	case accessUnauthorized:
-		Unauthorized("access denied").ServeHTTP(w, r)
-		return
-	case accessForbidden:
-		Forbidden("forbidden").ServeHTTP(w, r)
+	ucred, err := ucrednetGet(r.RemoteAddr)
+	if err != nil && err != errNoID {
+		logger.Noticef("Cannot parse UID from remote address %q: %s", r.RemoteAddr, err)
+		InternalError(err.Error()).ServeHTTP(w, r)
 		return
 	}
 
 	var rspf ResponseFunc
-	var rsp = MethodNotAllowed("method %q not allowed", r.Method)
+	var access AccessChecker
 
 	switch r.Method {
 	case "GET":
 		rspf = c.GET
+		access = c.ReadAccess
 	case "PUT":
 		rspf = c.PUT
+		access = c.WriteAccess
 	case "POST":
 		rspf = c.POST
-	case "DELETE":
```

Comment:
What happened to DELETE here?

Reply:
[Earlier](https://github.com/canonical/pebble/pull/358#discussion_r1488816769) I had suggested removing it as snapd removed it some time ago (and Pebble wasn't using it either). We can always add it back if we need it, but this was to be more in line with snapd.

Reply:
OK, happy for it to go, since it isn't used and inline with snapd.